packages/core/src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -109,7 +109,7 @@ const BottomSheet = React.forwardRef<BottomSheetComponent, BottomSheetProps>(
           borderWidth,
           borderColor: borderColor ?? theme.colors.border.brand,
         }}
-        onChange={(index) => onSettle?.(mappedSnapPoints.length - index - 1)}
+        onChange={(index) => onSettle?.(index)}
```

Comment:
I don't think this is the issue. Looking at the linear issue (and testing on my own), seems the bottom sheet has 4 snap points, even though by default there should only be 3 (top, middle, bottom). I think this inconsistency is what is leading to the incorrect index, but the existing logic for this is still correct.

Reply:
Seems there's a default bottommost snap point for some reason

Reply:
I can investigate this later today, but let me know if you have any idea why this could be happening.

Reply:
Ya, I will investigate more today 🙏 thanks for point out